iceoryx_hoofs/test/moduletests/test_container_vector.cpp

```diff
@@ -960,6 +963,22 @@ TEST_F(vector_test, EraseOfMiddleElementCallsDTorAndMove)
     EXPECT_THAT(moveAssignment, Eq(2U));
 }
 
+TEST_F(vector_test, AccessOfNonExistingElementWithAtLeadTermination)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "31a4f0fb-31dd-4269-9bec-31ef0542c42b");
+    EXPECT_THAT(sut.empty(), Eq(true));
+
+    IOX_EXPECT_FATAL_FAILURE<iox::HoofsError>([&] { sut.at(69); }, iox::HoofsError::EXPECTS_ENSURES_FAILED);
+}
+
+TEST_F(vector_test, AccessOfNonExistingElementWithBracketLeadTermination)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "69a4f0fb-3d31-4273-9bec-69ef05a4242b");
+    EXPECT_THAT(sut.empty(), Eq(true));
+
+    IOX_EXPECT_FATAL_FAILURE<iox::HoofsError>([&] { sut[31]; }, iox::HoofsError::EXPECTS_ENSURES_FAILED);
```

Comment:
Same as above, I would propose to check `size() + 1` and maybe some other index optionally, also depending on `size`.